migrations/helpers/dart_samples_update_helper.py

```diff
@@ -0,0 +1,387 @@
+
+        if number_docs_found > 0:
+            print(f"Updating MLWH database for {len(mongo_docs_for_sql)} sample documents")
+            # create connection to the MLWH database
+            with create_mysql_connection(config, False) as mlwh_conn:
+
+                # execute sql query to insert/update timestamps into MLWH
+                run_mysql_executemany_query(mlwh_conn, SQL_MLWH_MULTIPLE_INSERT, mongo_docs_for_sql)
+        else:
+            print("No documents found for this timestamp range, nothing to insert or update in MLWH")
+
+        # 7. add all plates to DART, but only RESULT=Positive samples
+        update_dart_filtered_positive_fields(config, samples)
+    except Exception as e:
+        print(e)
+
+
+def add_sample_uuid_field(samples: List[Sample]) -> List[Sample]:
+    # add lh sample uuid
+    for sample in samples:
+        if FIELD_LH_SAMPLE_UUID not in [*sample]:
+            sample[FIELD_LH_SAMPLE_UUID] = str(uuid.uuid4())
+
+    return samples
+
+
+def update_mongo_fields(mongo_db, samples: List[Sample], version: str, update_timestamp: datetime) -> bool:
+    """Bulk updates sample filtered positive fields in the Mongo database
+
+    Arguments:
+        config {ModuleType} -- application config specifying database details
+        samples {List[Sample]} -- the list of samples whose filtered positive fields should be updated
+        version {str} -- the filtered positive identifier version used
+        update_timestamp {datetime} -- the timestamp at which the update was performed
+
+    Returns:
+        bool -- whether the updates completed successfully
+    """
+    # get ids of those that are filtered positive, and those that aren't
+    all_ids: List[str] = [sample[FIELD_MONGODB_ID] for sample in samples]
+    filtered_positive_ids: List[str] = [
+        sample[FIELD_MONGODB_ID] for sample in list(filter(lambda x: x[FIELD_FILTERED_POSITIVE] is True, samples))
+    ]
+    filtered_negative_ids = [mongo_id for mongo_id in all_ids if mongo_id not in filtered_positive_ids]
+
+    samples_collection = get_mongo_collection(mongo_db, COLLECTION_SAMPLES)
+    samples_collection.update_many(
+        {FIELD_MONGODB_ID: {"$in": filtered_positive_ids}},
+        {
+            "$set": {
+                FIELD_FILTERED_POSITIVE: True,
+                FIELD_FILTERED_POSITIVE_VERSION: version,
+                FIELD_FILTERED_POSITIVE_TIMESTAMP: update_timestamp,
+            }
+        },
+    )
+    samples_collection.update_many(
+        {FIELD_MONGODB_ID: {"$in": filtered_negative_ids}},
+        {
+            "$set": {
+                FIELD_FILTERED_POSITIVE: False,
+                FIELD_FILTERED_POSITIVE_VERSION: version,
+                FIELD_FILTERED_POSITIVE_TIMESTAMP: update_timestamp,
+            }
+        },
+    )
+    return True
+
+
+def samples_updated_with_source_plate_uuids(mongo_db, samples: List[Sample]) -> List[Sample]:
```

Comment:
Would it be difficult to pull this logic shared with the `file_processing.py` equivalent out to a shared location? The different types or logging/error reporting may make this a bit fiddly

Reply:
It probably wouldn't be too difficult but thought of leaving it for a larger refactor to move migrations "closer" to the main codebase. Currently it skirts the main `runner.py` but ideally we should be calling migrations using a sub-parser for argparse, i.e. `python runner.py migrate <name>`